core/src/main/scala/org/apache/spark/deploy/security/HadoopFSDelegationTokenProvider.scala

```diff
@@ -103,15 +103,17 @@ private[deploy] class HadoopFSDelegationTokenProvider(fileSystems: Configuration
 
   private def getTokenRenewalInterval(
       hadoopConf: Configuration,
-      filesystems: Set[FileSystem]): Option[Long] = {
+      filesystems: Set[FileSystem],
+      creds:Credentials): Option[Long] = {
     // We cannot use the tokens generated with renewer yarn. Trying to renew
     // those will fail with an access control issue. So create new tokens with the logged in
     // user as renewer.
-    val creds = fetchDelegationTokens(
+    val fetchCreds = fetchDelegationTokens(
```

Comment:
Also here the diff in spark2.2 and master
=> is missing PRINCPAL(aka spark.yarn.principal) config. Not sure if we need to do this now. Let me know your opinion @vanzin @tgravescs 

sparkConf.get(PRINCIPAL).flatMap { renewer =>
      val creds = new Credentials()
      hadoopFSsToAccess(hadoopConf, sparkConf).foreach { dst =>
        val dstFs = dst.getFileSystem(hadoopConf)
        dstFs.addDelegationTokens(renewer, creds)
      }

Reply:
That code was in `getTokenRenewalInterval`; that call is only needed when principal and keytab are provided, so adding the code back should be ok. It shouldn't cause any issues if it's not there, though, aside from a wasted round trip to the NNs.

Reply:
I'd prefer to not call it if we don't need to so as long as adding the config back doesn't mess with the mesos side of things (since this is now common code) I think that would be good.  the PRINCIPAL config is yarn specific config, but looking at SparkSubmit it appears to be using for mesos as well.  

@vanzin do you happen to know if mesos is using that as well, I haven't kept up with mesos kerberos support.  so not sure if more is going to happen there.

Reply:
I'm pretty sure Mesos is not currently hooked up to the principal / keytab stuff. It just picks up the initial delegation token set, and when those expire, things stop working.

Adding the check back here is the right thing; it shouldn't affect Mesos when it adds support for principal / keytab (or if it does, it can be fixed at that time).